src/Compilers/Core/Portable/SourceGeneration/SourceGeneratorContext.cs

```diff
@@ -41,6 +42,11 @@ internal SourceGeneratorContext(Compilation compilation, ImmutableArray<Addition
         /// </remarks>
         public Compilation Compilation { get; }
 
+        /// <summary>
+        /// Get the <see cref="ParseOptions"/> that will be used to parse any added sources.
+        /// </summary>
+        public ParseOptions ParseOptions { get; }
```

Comment:
Can you add tests for the added public API?

Also, just for my context, can you clarify where those parse options come from (how are they computed)?

Reply:
Added a test. The options are passed to the generator driver at construction time. In CSC we pass them here: https://github.com/dotnet/roslyn/blob/5a609283e57b2f1ffc1011a1467afe1edf5ff2cd/src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs#L961 

Which come directly from the command line args.